tensorflow_probability/python/experimental/mcmc/particle_filter.py

```diff
@@ -474,7 +585,8 @@ def propose_and_update_log_weights_fn(step, state, seed=None):
           particles=proposed_particles,
           log_weights=log_weights + _compute_observation_log_weights(
```

Comment:
It looks like `_compute_observation_log_weights` needs to be updated to add a `particles_dim` argument.

And that function will have to do something like:
```
observation = tf.expand_dims(observation, axis=particles_dim)
```
so that the computed log-weights have the right shape.